substratevm/src/com.oracle.svm.jni/src/com/oracle/svm/jni/access/JNIAccessFeature.java

```diff
@@ -184,28 +184,48 @@ private static ConditionalConfigurationRegistry getConditionalConfigurationRegis
         return (ConditionalConfigurationRegistry) ImageSingletons.lookup(JNIRuntimeAccess.JNIRuntimeAccessibilitySupport.class);
     }
 
-    private static JNICallTrampolineMethod createJavaCallTrampoline(BeforeAnalysisAccessImpl access, CallVariant variant, boolean nonVirtual) {
-        MetaAccessProvider wrappedMetaAccess = access.getMetaAccess().getWrapped();
-        ResolvedJavaField field = JNIAccessibleMethod.getCallWrapperField(wrappedMetaAccess, variant, nonVirtual);
+    private void createJavaCallTrampoline(BeforeAnalysisAccessImpl access, CallVariant variant, boolean nonVirtual) {
+        AnalysisMetaAccess metaAccess = access.getMetaAccess();
+        ResolvedJavaField field = JNIAccessibleMethod.getCallWrapperField(metaAccess.getWrapped(), variant, nonVirtual);
         access.getUniverse().lookup(field.getDeclaringClass()).registerAsReachable();
         access.registerAsAccessed(access.getUniverse().lookup(field));
```

Comment:
I think you don't need the wrapped meta access here either.

Reply:
It looks like I do, removing it results in:

```
Fatal error: java.lang.ClassCastException: class jdk.vm.ci.hotspot.HotSpotResolvedJavaMethodImpl cannot be cast to class com.oracle.graal.pointsto.meta.AnalysisMethod (jdk.vm.ci.hotspot.HotSpotResolvedJavaMethodImpl is in module jdk.internal.vm.ci of loader 'bootstrap'; com.oracle.graal.pointsto.meta.AnalysisMethod is in unnamed module of loader 'app')
	at com.oracle.graal.pointsto.meta.AnalysisType.resolveConcreteMethod(AnalysisType.java:844)
	at com.oracle.graal.pointsto.meta.AnalysisType.resolveConcreteMethod(AnalysisType.java:73)
	at com.oracle.graal.pointsto.meta.AnalysisType.resolveConcreteMethod(AnalysisType.java:851)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.collectMethodImplementations(AnalysisUniverse.java:622)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.collectMethodImplementations(AnalysisUniverse.java:612)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.collectMethodImplementations(AnalysisUniverse.java:612)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.collectMethodImplementations(AnalysisUniverse.java:612)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.collectMethodImplementations(AnalysisUniverse.java:612)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.collectMethodImplementations(AnalysisUniverse.java:612)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.getMethodImplementations(AnalysisUniverse.java:596)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.collectMethodImplementations(AnalysisUniverse.java:583)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.setAnalysisDataValid(AnalysisUniverse.java:167)
	at com.oracle.graal.pointsto.PointsToAnalysis.finish(PointsToAnalysis.java:651)
	at com.oracle.graal.pointsto.PointsToAnalysis.runAnalysis(PointsToAnalysis.java:734)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:701)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:536)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:494)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:426)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:587)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:126)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:617)
```